Rx.NET/Documentation/Rx-Roadmap-2023.md

```diff
@@ -0,0 +1,170 @@
+
+We have produced a prototype demonstrating what Rx might look like if the UWP parts were separated: https://github.com/idg10/reactive/pull/3. Since this removes the UWP target from `System.Reactive`, this includes a breaking change for UWP clients: they will find that the `ThreadPoolScheduler` now looks identical to the ones available to all other Rx clients; if they need the old functionality it is available through a new `WindowsThreadPoolScheduler` in the `System.Reactive.Windows.Threading` library, which is identical to the old UWP-specific `ThreadPoolScheduler` in everything but name.
+
+
+## Plans for upcoming release
+
+We need to decide whether we should try to solve all of the problems in one step, or whether we produce any interim releases that get us part way to a solution. Our current view is that we will not attempt to do everything at once. We would like to get automated tests running against .NET 6.0 and .NET 7.0 in place as soon as possible so that we can verify that these platforms are supported. (This necessarily entails fixing the problems that prevent Rx from being built using the latest tool versions.) And we would like to make preview builds of AsyncRx.NET available as quickly as possible.
+
+The issues around UI framework dependencies are arguably trickier, because fixing these will necessarily be more disruptive. Fixing the code bloat problems will mean ensuring that projects targeting some Windows-specific TFM won't automatically get the WPF and Windows Forms support just because they use Rx. This seems like it would need to be a breaking change (unless we moved all of Rx into some completely new NuGet packages, and turned the existing `System.Reactive` package into a façade offered only for backwards compatibility). Moreover, there are various subtle issues surrounding this, and we would need to be certain that we were not causing any regressions. For example, we would need to be certain that we were not re-introducing the old bug (https://github.com/dotnet/reactive/issues/97) in which plug-in hosts (notably Visual Studio) could load different plug-ins that had different opinions over whether to load the `netstandard2.0` or the `net4x` flavour or Rx, resulting in unresolvable conflicts and runtime errors. We would also need to avoid re-tracing the steps that caused the problems in https://github.com/dotnet/reactive/issues/199#issuecomment-266138120. We don't want to delay everything else while we experiment with strategies for this problem. Moreover, there are now numerous .NET frameworks (including, but not limited to Avalonia, Unity, Xamarin, WinUI) that we would want to be confident of working well with. So although we want to resolve these particular problems as soon as possible, we think this will take time, and don't want to delay the other goals.
+
+### V.next (v5.1?)
```

Comment:
Do you think 5.1 is enough of a version increment for a set of breaking changes?

Reply:
I don't know. I'm on the fence.

On the one hand, this is a statement of intent: this is a minor change. The only breaking change I'm aware of is that it no longer works on old unsupported runtimes. If you are operating in a world of currently supported runtimes, then this does not (as far as I am aware) constitute a breaking change.

On the other hand, it does seem a little odd to drop some target frameworks without making it a major version bump. But to call this 6.0 seems to me to be overpromising slightly.

(And there's also the irritating fact that Rx's version numbers accidentally aligned with .NET's, leading to some misunderstandings. I quite like the idea of taking the opportunity to break that false connection.)

Reply:
We could always make a comment that one of our decisions is to move towards semantic versioning (as per our endjin IP), which means with the changes we're going to make we'll quickly make a number of breaking changes and go 6.0 / 7.0 / 8.0 quite rapidly. You could even do something intentionally bold and jump to a higher number like 9.0 or 10.0